booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,654 @@
+    "Validate documentParties",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var documentParties = body.path("documentParties");
+      var partyFunctions = StreamSupport.stream(documentParties.spliterator(), false)
+        .map(p -> p.path("partyFunction"))
+        .filter(JsonNode::isTextual)
+        .map(n -> n.asText(""))
+        .collect(Collectors.toSet());
+
+      if (!partyFunctions.contains("SCO")) {
+        if (!body.path("serviceContractReference").isTextual()) {
+          issues.add("The 'SCO' party is mandatory when 'serviceContractReference' is absent");
+        }
+        if (!body.path("contractQuotationReference").isTextual()) {
+          issues.add("The 'SCO' party is mandatory when 'contractQuotationReference' is absent");
+        }
+      }
+      Set<String> partyShippers  = new HashSet<>(Arrays.asList("OS", "DDR"));
+      if (partyFunctions.stream().noneMatch(partyShippers::contains)) {
+        if (!partyFunctions.contains("BA")) {
+          issues.add("The 'BA' party must be provided if 'OS' or 'DDR' are absent ");
+        }
+      }
+        return issues;
+    });
+
+  private static final List<JsonContentCheck> STATIC_BOOKING_CHECKS = Arrays.asList(
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/cargoMovementTypeAtOrigin"), BookingDataSets.CARGO_MOVEMENT_TYPE),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/cargoMovementTypeAtDestination"), BookingDataSets.CARGO_MOVEMENT_TYPE),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/communicationChannelCode"), BookingDataSets.COMMUNICATION_CHANNEL_CODES),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/declaredValueCurrency"), BookingDataSets.ISO_4217_CURRENCY_CODES),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/incoTerms"), BookingDataSets.INCO_TERMS_VALUES),
+    JsonAttribute.atMostOneOf(
+      JsonPointer.compile("/shippedOnBoardDate"),
+      JsonPointer.compile("/receivedForShipmentDate")
+    ),
+    CHECK_EXPECTED_DEPARTURE_DATE,
+    CHECK_EXPECTED_ARRIVAL_POD,
+    NOR_PLUS_ISO_CODE_IMPLIES_ACTIVE_REEFER,
+    IS_SHIPPER_OWNED_CONTAINER,
+    REFERENCE_TYPE_VALIDATION,
+    ISO_EQUIPMENT_CODE_VALIDATION,
+    IS_EXPORT_DECLARATION_REFERENCE_PRESENT,
+    IS_IMPORT_DECLARATION_REFERENCE_PRESENT,
+    OUTER_PACKAGING_CODE_IS_VALID,
+    TLR_CC_T_COMBINATION_VALIDATIONS,
+    DOCUMENT_PARTY_FUNCTIONS_MUST_BE_UNIQUE,
+    UNIVERSAL_SERVICE_REFERENCE,
+    VALIDATE_SHIPMENT_CUTOFF_TIME_CODE,
+    VALIDATE_ALLOWED_SHIPMENT_CUTOFF_CODE,
+    VALIDATE_DOCUMENT_PARTIES,
+    JsonAttribute.atLeastOneOf(
+      JsonPointer.compile("/expectedDepartureDate"),
+      JsonPointer.compile("/expectedArrivalAtPlaceOfDeliveryStartDate"),
+      JsonPointer.compile("/expectedArrivalAtPlaceOfDeliveryEndDate"),
+      JsonPointer.compile("/carrierExportVoyageNumber")
+    ),
```

Comment:
I see you changed this into a `atLeastOneOf`. Did you clarify with Flavia whether this is truly a `atLeastOneOf` situation?  (I think the earlier version of code I commented on turned out not to be a simple `one of` and I had to amend my remark when I re-read the code)

Reply:
I checked with Henrik and he confirmed that atleast any of these fields must be present.